tckdb/backend/app/utils/doi_lookup.py

```diff
@@ -0,0 +1,52 @@
+from typing import Dict, Optional
+
+import requests
+
+BASE_URL = "https://api.crossref.org/works/"
+
+
+def fetch_doi_metadata(doi: str) -> Optional[Dict[str, any]]:
```

Comment:
this is a long shot, but if this api is not ours we will need at some point make sure we are not over-questioning him and making him ban us. might not be a problem at all, just pointing out :)

Reply:
https://api.crossref.org/swagger-ui/index.html

So, there is a rate limit but I don't think it will be a problem for us currently. Although, we do need to adjust the `mailto` so that they have contact information (required by them)

Reply:
And regarding the blocking, it seems you just get blocked for 10 seconds and there is an error code of 429. We should make an exception for that and if we get it too many times we back off completely.

Reply:
ok sounds like a good solution, we can create a ticket for it if it's not urgent for now